src/torchjd/aggregation/bases.py

```diff
@@ -27,10 +27,14 @@ def _check_is_finite(matrix: Tensor) -> None:
 
     @abstractmethod
     def forward(self, matrix: Tensor) -> Tensor:
-        raise NotImplementedError
+        """Computes and returns the aggregation from the input matrix."""
 
-    # Override to make type hints more specific
+    # Override to make type hints and documentation more specific
     def __call__(self, matrix: Tensor) -> Tensor:
+        """
+        Applies all registered hooks and computes and returns the aggregation from the input matrix.
```

Comment:
Again, I would argue for removing the "and returns", the function computes the aggregation and applies the hooks, so something like this?

```Computes the aggregation of the input `matrix` and applies all registered hooks.```

The same applies identically below (I let you decide, the current formulation is fine if you prefer it)